q2cli/info.py

```diff
@@ -6,41 +6,41 @@
 # The full license is in the file COPYING.txt, distributed with this software.
 # ----------------------------------------------------------------------------
 
-
-import sys
-
-import pip
 import click
-import qiime
-
-from . import __version__ as q2cli_version
 
 
 def echo_version(ctx=None, name=None, value=True):
     if value:
+        # Conditional imports because this callback appears to be invoked when
+        # any command is run (in most cases, `value` is `False`).
+        import sys
+        import qiime
+        import q2cli
+
         pyver = sys.version_info
         click.echo('Python version: %d.%d.%d' %
                    (pyver.major, pyver.minor, pyver.micro))
         click.echo('QIIME version: %s' % qiime.__version__)
-        click.echo('q2cli version: %s' % q2cli_version)
+        click.echo('q2cli version: %s' % q2cli.__version__)
 
 
 def _echo_plugins():
-    plugin_manager = qiime.sdk.PluginManager()
-    installed_plugins = plugin_manager.plugins
-    if len(installed_plugins) == 0:
-        # TODO: update this URL, pending:
-        # https://github.com/qiime2/qiime2/issues/80
-        # https://github.com/qiime2/qiime2/issues/82
+    import qiime.sdk
+    import q2cli.cache
+
+    plugins = q2cli.cache.CACHE.plugins
+    if plugins:
+        for name, plugin in sorted(plugins.items()):
+            click.echo('%s %s' % (name, plugin['version']))
+    else:
         click.secho('No plugins are currently installed.\nYou can browse '
                     'the official QIIME 2 plugins at: '
-                    'http://2.qiime.org/Plugins')
-    else:
-        for name, plugin in installed_plugins.items():
-            click.echo('%s %s' % (name, plugin.version))
+                    '%s/Plugins' % qiime.sdk.HELP_URL)
 
 
 def _echo_installed_packages():
+    import pip
```

Comment:
Not important for this PR, but we should switch to using `pkg_resources.working_set` for this.